engine/polyglot-api/src/main/java/org/enso/polyglot/debugger/IdExecutionService.java

```diff
@@ -10,6 +10,24 @@ public interface IdExecutionService {
   String INSTRUMENT_ID = "id-value-extractor";
 
   public interface Callbacks {
+
+    void onEnter(UUID nodeId);
+
+    void onExpressionReturn(UUID nodeId, Object result, long nanoElapsedTime);
+
+    /**
+     * Notification when a returned value is a function.
+     *
+     * @param nodeId identification of the node to be computed
+     * @param result info about function call
+     * @return {@code null} should the execution of the node be performed; any other value to skip
+     *     the execution and return the value as a result.
+     */
+    Object onFunctionReturn(UUID nodeId, TruffleObject result);
+  }
+
+  public interface ExecutionSupport {
```

Comment:
I don't understnand why we should have yet another interface - `ExecutionSupport` - no, there is only one `Callbacks` interface and that is _the interface_. Unless _really justified_, I want to keep this _interface_ simple and not convoluted.

Reply:
I thought it's a good idea to separate the callbacks and the logic around the execution. Anyway, I reverted the change